class.photon.php

```diff
@@ -388,6 +389,24 @@ public static function filter_the_galleries( $galleries ) {
 		return $galleries;
 	}
 
+
+	/**
+	 * Runs the image widget through photon.
+	 *
+	 * @param array $instance Image widget instance data.
+	 * @return array
+	 */
+	function filter_the_image_widget( $instance ) {
+		if ( Jetpack::is_module_active( 'photon' ) && ! $instance['attachment_id'] && $instance['url'] ) {
+			jetpack_photon_url( $instance['url'], array(
```

Comment:
Cross-posting from https://github.com/xwp/wp-core-media-widgets/issues/43#issuecomment-300324231

In regards to adding a new filter:

> I think the issue is that Photon works best when it has access to the underlying `attachment_id`, `width`, and `height` which is what `Jetpack_Photon::filter_the_content()` is able to parse out of the `img` tag's `class` attribute, correct? To me it seems that `Jetpack_Photon::filter_the_content()` should be broken up into a couple methods to support the situation where all of the props are pre-parsed for you, as they would be in this case here. The image widget `$instance` has the size-specific image `url`, the `attachment_id`, the `width`, the `height`, and the selected image `size`. All of these pieces should be able to be used by Photon to create the best image URL without any need to re-parse HTML.